### PR TITLE
First commit for node assignments

### PR DESCRIPTION
**Pull request by prince594**
@anjanaNewput 
Please review and approve.

### CODE REVIEW

File: nodeTs/src/config/config.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const MONGO_OPTIONS = {
+    useUnifiedTopology: true,
+    useNewUrlParser: true,
+    socketTimeoutMS: 30000,
+    keepAlive: true,
+    autoIndex: false,
+    retryWrites: false
+};
+
+const MONGO_USERNAME = process.env.MONGO_USERNAME || 'princeNewput';
```
Comment (anjanaNewput) [inline nodeTs/src/config/config.ts]: never commit your username and password.

File: nodeTs/src/controllers/books.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    let patchRes: any = await services.patchById(bookId);
+    return res.send(patchRes);
+  } catch (error: any) {
+    res.send(error);
+  }
+};
+/* GET */
+const getBooks = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    console.log(req.query);
```
Comment (anjanaNewput) [inline nodeTs/src/controllers/books.ts:30]: remove console

File: nodeTs/src/controllers/user.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+const login = async (req: any, res: Response, next: NextFunction) => {
+  try {
+    const { email, password } = req.body;
+    if (!email || !password) {
+      next(console.log("Please provide email and password"));
+      res.send({ message: "Please provide email and password" });
+    }
+    const userData: any = await userServices.getUserByEmail(email);
+    if (!userData) {
+      next(console.log("Invalid user"));
```
Comment (anjanaNewput) [inline nodeTs/src/controllers/user.ts]: why we need this

File: nodeTs/src/services/books.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import Book from "../models/books";
+/* GET */
+const finder = async (bookId: any) => {
+  try {
+    if (!bookId) {
```
Comment (anjanaNewput) [inline nodeTs/src/services/books.ts]: we can improve it like const query = {}; if(bookid) {query['id'] = bookId }
 const results = await Book.find(query).exec();
      return results;

File: nodeTs/src/controllers/user.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      res.send({ message: "Please provide valid user credential" });
```
Comment (anjanaNewput) [inline nodeTs/src/controllers/user.ts]: move messages in constant file

File: nodeTs/src/controllers/user.ts (hunk 3 above)
Comment (prince594) [reply nodeTs/src/controllers/user.ts]: This for login route, and before login server checks if the user is already existing in db.

**Pull request merged**